### ISSUE TITLE
Nested "Meta" class deleted after class is created

### ISSUE THREAD

**Issue by j2kun** [elastic/elasticsearch-dsl-py#660]
The `DocType` base class deletes a nested `Meta` class after class creation, which is counterintuitive. I want to be able to access the data defined there in a generic base class.

```
In [1]: from elasticsearch_dsl import DocType
   ...: 

In [2]: class A(DocType):
   ...:     class Meta:
   ...:         x = 'wat'
   ...:         

In [3]: A.Meta
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-a3b6902bd030> in <module>()
----> 1 A.Meta

AttributeError: type object 'A' has no attribute 'Meta'
```

A "normal" example of a nested class.

```
In [4]: class B():
   ...:     class Meta:
   ...:         x = 'wat'
   ...:         

In [5]: B.Meta
Out[5]: __main__.B.Meta
```

**Comment by j2kun** [comment]
My current workaround is to look at the created `_doc_type` object on the class, but nontheless this is counterintuitive and took me a while to find. In particular, searching this repository for `Meta` produce no actionable information as to why this class was being deleted.

**Comment by j2kun** [comment]
I just wanted to access the `index` attribute, which I'm using `_doc_type` for now. My concern, and what took time for me to resolve, is that this just violates the principle of least surprise: the docs define a `Meta` in the starter example, this behavior is undocumented, I can't find where the change happens in the source code, and you're now telling me the best practice is to access `index` through a protected member variable.

If you don't want to change the code, updating the documentation would help. But "limiting the namespace" seems like a nonissue if the user sees that they're defining it. Alternatively, instead of `Meta` call it something more specific like `DocTypeMeta` so it wouldn't collide with more the common `Meta`. Alternatively alternatively, let the user call `Meta` whatever they want, and have them decorate it so you know which class to use to create `_doc_type`, and they can be responsible for namespace collisions.

Food for thought :)

**Comment by HonzaKral** [comment]
For the `index` attribute going trhough the `_doc_type` is the proper documented solution. I will have a look how to make it more prominent. It is protected as it contains the internal state like the `index` name. This is btw a pattern that I lifted from the Django ORM and many people are very used to it.

Thanks for raising the issue, but since there is no good solution (if we kept `Meta` visible then it would still be a midway solution because of all the field objects) and it is a well established pattern I will keep it as is.
